downloader/src/main/kotlin/Downloader.kt

```diff
@@ -245,7 +245,7 @@ object Downloader {
             url = target.vcsProcessed.url,
             revision = target.vcsProcessed.revision.takeIf { it.isNotBlank() } ?: revision,
             resolvedRevision = revision,
-            path = target.vcsProcessed.path // TODO: Needs to check if the VCS used the sparse checkout.
+            path = target.vcsProcessed.path
```

Comment:
I understand that comment so that it could be that the VCS ignored the path, e.g. because sparse checkouts are not supported, and that it should be set to blank here then. But to implement that I think it would be better if `download` would return a `VcsInfo` instead of building it here.

Reply:
Well, `VersionControlSystem.download()` returns a `WorkingTree` that can be queried for its `VcsInfo`. But the semantics of the `path` in *that* `VcsInfo` are different from the `path` that we return in the `VcsInfo` as part of the `DownloadResult`:

- The `path` of the `VcsInfo` in the `WorkingTree` contains the relative `workingDir` to the VCS root.
- The `path` of the `VcsInfo` in the `DownloadResult` contains the sparse checkout / manifest path, if any.

That's exactly what I was confused about myself, and what I'm trying to clarify as part of this PR.

That said, I'm not entirely happy about these semantics, and maybe we should try to change them. But the question is how. Making the `VcsInfo` in the `WorkingTree` return the sparse checkout / manifest path, if any, is quite hard to implement generically and for each VCS. Aligning semantics the other way around by making the `VcsInfo` in the `DownloadResult` return the relative `workingDir` to the VCS root is much easier, but it misses the point of maintaining the information about what was downloaded. So it seems we cannot easily get rid of these split semantics.